src/choose-location/choose-location.component.tsx

```diff
@@ -1,44 +1,40 @@
 import React from "react";
-import { always } from "kremling";
+import { RouteComponentProps, StaticContext } from "react-router";
 import { useConfig } from "@openmrs/esm-module-config";
-import { createErrorHandler } from "@openmrs/esm-error-handling";
 import { getCurrentUser } from "@openmrs/esm-api";
-import { Trans } from "react-i18next";
-import {
-  setSessionLocation,
-  searchLocationsFhir,
-} from "./choose-location.resource";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { setSessionLocation, queryLocations } from "./choose-location.resource";
+import { LocationEntry } from "../types";
+import LocationPicker from "../location-picker/location-picker.component";
 import navigate from "../navigate";
-import styles from "../styles.css";
-import { debounce, isEmpty } from "lodash";
+
+interface LoginReferrer {
+  referrer?: string;
+}
+
+interface ChooseLocationProps
+  extends RouteComponentProps<{}, StaticContext, LoginReferrer> {}
 
 export default function ChooseLocation(props: ChooseLocationProps) {
+  const referrer = props.location?.state?.referrer;
+
   const config = useConfig();
-  const [locationData, setLocationData] = React.useState({
-    activeLocation: "",
-    locationResult: [],
-    emptyResult: false,
-  });
-  const [searchTerm, setSearchTerm] = React.useState("");
-  const [isSubmitting, setIsSubmitting] = React.useState(false);
-  const [errorMessage, setErrorMessage] = React.useState("");
-  const locationInputRef = React.useRef<HTMLInputElement>(null);
-  const formRef = React.useRef<HTMLFormElement>(null);
+  const [loginLocations, setLoginLocations] = React.useState<
+    Array<LocationEntry>
+  >(null);
   const [currentUser, setCurrentUser] = React.useState(null);
-  const searchTimeout = 300;
-  React.useEffect(() => {
-    getCurrentUser().subscribe((user) => {
-      setCurrentUser(user ? user.display : currentUser);
-    }, createErrorHandler());
-  });
+  const [loading, setLoading] = React.useState(true);
```

Comment:
Can we keep the `is` prefix for booleans? `isLoading`, `setIsLoading`?

Reply:
Sure!